src/clims/models/container.py

```diff
@@ -17,13 +17,15 @@ class Container(ExtensibleModel):
     parent = FlexibleForeignKey('Container', null=True, related_name="children_set")
 
     # TODO: Benefits of FlexibleForeignKey
-    # TODO: Decide on naming for related_name
     container_type = FlexibleForeignKey('ContainerType', null=True, related_name="containers")
 
     name = models.TextField(null=True)
 
     organization = FlexibleForeignKey('sentry.Organization')
 
+    # The work batch in which this container was created, if it was created in a work batch context.
+    work_batch = FlexibleForeignKey("WorkBatch", null=True, related_name="output_containers")
```

Comment:
The association between container and workbatch is defined in two ways, 1) from container to workbatch, 2) from container to workunit to workbatch. In the end, I think this might lead to confusing code. Imagine that assiciation 2) is used almost all the time, except in some corner cases where association 1) is used. I can smell bugs coming here.

Reply:
Right. This is just a pointer to the containers that were created within the workbatch. We'll need them for the UI to be able to show which containers are shown on the right in the Transition component.

The containers in WorkUnits are the input containers, but only in the case if someone is writing a workflow that tracks containers rather than substances.

I think that the dangers you're talking about are difficult to guard against more while data modelling. If the link is required we'll have to have it. But we can use naming to help, and I think we are doing that because the link in the work_batch is called `output_containers`. We can try to guard against wrong usage more by having good documentation and a clear API in the service layer.

Reply:
Now I see that this association is for the output containers, and the other is for the input containers, and that changes the situation a bit. Probably Camunda is only tracking input samples/containers, right? I still think, however, that this violates the "principle of least surprises".